opal/corelib/comparable.rb

```diff
@@ -80,7 +80,14 @@ def between?(min, max)
     true
   end
 
-  def clamp(min, max)
+  def clamp(min, max=nil)
+    if Range === min && max.nil?
+      max = min.end
+      max = Float::INFINITY if max.nil?
+      min = min.begin
+      min = -Float::INFINITY if min.nil?
```

Comment:
Can this be achieved with a simple `||`? I don't think we should consider `false` as a possible valid range edge

```rb
max = min.end || Float::INFINITY
min = min.being || -Float::INFINITY
```

Reply:
false is a possible value and doesn't mean it's an infinite range

Reply:
i refactored this part anyway :)

Reply:
No, please if it's a possible range it's fine the way it is 👍 I just wasn't able to construct a range with a boolean, until now, it worked with `false..false`